apps/prairielearn/src/middlewares/authzCourseOrInstance.js

```diff
@@ -323,6 +331,10 @@ module.exports = asyncHandler(async (req, res, next) => {
   // we simply return (without error). This allows the authn user to keep access
   // to pages (e.g., the effective user page) for which only authn permissions
   // are required.
+  //
+  // TODO: this might require a change. The `WHERE` clause changed from looking
+  // at `has_student_access_with_enrollment` to just `has_student_access`.
+  // We should check here if we have to automatically enroll the student?
```

Comment:
I left this comment during an earlier iteration of the code. The "automatically enroll the student" part is definitely outdated, but I'd like to make sure we check that I'm correctly handling the "has access but is not enrolled" case.